src/DataForm/DataForm.php

```diff
@@ -377,6 +377,11 @@ protected function process()
                     $this->process_status = "success";
                 }
                 foreach ($this->fields as $field) {
+                    //Field may be setted on other mode like "readonly", should not update it's value.
+                    if ($field->mode != 'editable') {
```

Comment:
And this can also avoid data hacking ( change http post data ). 
If a column is on readonly mode, it should not do modify.